actors/miner/tests/prove_commit.rs

```diff
@@ -340,3 +354,343 @@ fn prove_sectors_from_batch_pre_commit() {
+    rt.set_epoch(precommit_epoch + max_prove_commit_duration(&rt.policy, h.seal_proof_type) - 1);
+
+    h.prove_commit_sector(&mut rt, &pre_commit_a, h.make_prove_commit_params(sector_no_a)).unwrap();
+    h.prove_commit_sector(&mut rt, &pre_commit_b, h.make_prove_commit_params(sector_no_b)).unwrap();
+
+    let conf = ProveCommitConfig {
+        verify_deals_exit: HashMap::from([(sector_no_a, ExitCode::USR_ILLEGAL_ARGUMENT)]),
+    };
+    h.confirm_sector_proofs_valid(&mut rt, conf, vec![pre_commit_a, pre_commit_b]).unwrap();
+    check_state_invariants(&rt);
+}
+
+#[test]
+fn prove_commit_just_after_period_start_permits_post() {
+    let h = ActorHarness::new(PERIOD_OFFSET);
+    let mut rt = h.new_runtime();
+    rt.balance.replace(TokenAmount::from(BIG_BALANCE));
+
+    // Epoch 101 should be at the beginning of the miner's proving period so there will be time to commit
+    // and PoSt a sector.
+    rt.set_epoch(101);
+    h.construct_and_verify(&mut rt);
+
+    // Commit a sector the very next epoch
+    rt.set_epoch(102);
+    let sector =
+        h.commit_and_prove_sector(&mut rt, MAX_SECTOR_NUMBER, DEFAULT_SECTOR_EXPIRATION, vec![]);
+
+    // advance cron to activate power.
+    h.advance_and_submit_posts(&mut rt, &[sector]);
+    check_state_invariants(&rt);
+}
+
+#[test]
+fn sector_with_non_positive_lifetime_is_skipped_in_confirmation() {
+    let h = ActorHarness::new(PERIOD_OFFSET);
+    let mut rt = h.new_runtime();
+    rt.balance.replace(TokenAmount::from(BIG_BALANCE));
+
+    let precommit_epoch = PERIOD_OFFSET + 1;
+    rt.set_epoch(precommit_epoch);
+
+    h.construct_and_verify(&mut rt);
+    let deadline = h.deadline(&rt);
+
+    let sector_no = 100;
+    let params = h.make_pre_commit_params(
+        sector_no,
+        precommit_epoch - 1,
+        deadline.period_end() + DEFAULT_SECTOR_EXPIRATION * rt.policy.wpost_proving_period,
+        vec![],
+    );
+    let precommit = h.pre_commit_sector(&mut rt, params, PreCommitConfig::empty(), true);
+
+    // precommit at correct epoch
+    rt.set_epoch(rt.epoch + rt.policy.pre_commit_challenge_delay + 1);
+    h.prove_commit_sector(&mut rt, &precommit, h.make_prove_commit_params(sector_no)).unwrap();
+
+    // confirm at sector expiration (this probably can't happen)
+    rt.set_epoch(precommit.info.expiration);
+    // sector skipped but no failure occurs
+    h.confirm_sector_proofs_valid(&mut rt, ProveCommitConfig::empty(), vec![precommit.clone()])
+        .unwrap();
+
+    // it still skips if sector lifetime is negative
+    rt.set_epoch(precommit.info.expiration + 1);
+    h.confirm_sector_proofs_valid(&mut rt, ProveCommitConfig::empty(), vec![precommit.clone()])
+        .unwrap();
+
+    // it fails up to the miniumum expiration
+    rt.set_epoch(precommit.info.expiration - rt.policy.min_sector_expiration + 1);
+    h.confirm_sector_proofs_valid(&mut rt, ProveCommitConfig::empty(), vec![precommit.clone()])
+        .unwrap();
+    check_state_invariants(&rt);
```

Comment:
Add the check that `st.get_sector(sector_no)` comes back empty/in error here as well

Reply:
Done.